icu4c/source/i18n/numparse_affixes.cpp

```diff
@@ -152,8 +157,12 @@ IgnorablesMatcher& AffixTokenMatcherWarehouse::ignorables() {
     return fSetupData->ignorables;
 }
 
-NumberParseMatcher& AffixTokenMatcherWarehouse::nextCodePointMatcher(UChar32 cp) {
-    return *fCodePoints.create(cp);
+NumberParseMatcher* AffixTokenMatcherWarehouse::nextCodePointMatcher(UChar32 cp, UErrorCode& status) {
+    auto* result = fCodePoints.create(cp);
```

Comment:
The way I understand ICU4C error handling, you shouldn't be attempting to create a new object unless U_SUCCESS(status). Isn't that so?

Reply:
Fixed.  Although, the way I understand ICU4C error handling, which very well could be wrong, is that there is no harm in calling new object if status is a failure, unless you are optimizing for the case of saving an extra potential call to malloc in the edge case of memory allocation failure.